src/storage/StorageModule.ts

```diff
@@ -1,32 +1,319 @@
-import { BigNumberish, ethers } from "ethers";
+import { BigNumberish, ethers, BytesLike, toBeHex, hexlify } from "ethers";
 import { IStorageModule } from "./IStorageModule";
+import { Codec, EvmUtils, Type } from "@/utils";
+import {
+    StateSnapshotStruct,
+    BalanceStruct,
+    BlockConfirmationStruct,
+    ExitChannelBlockStruct
+} from "@typechain-types/contracts/V1/DataTypes";
+import { UnrolledBlock, UnrolledSignedBlock } from "@/types/storage";
 
 export class StorageModule implements IStorageModule {
-    getLatestJoinChannelBlockHash(): string {
+    //BlockConfirmationStruct => SignedBlockStruct => encodedBlock => BlockStruct
+    private blockConfirmationStructsMap: Map<string, BlockConfirmationStruct>;
+    private latestBlockConfirmationKey: string;
+
+    // map [fork count, block height] => struct
+    // we make the key a string to avoid double mapping
+    private stateSnapshotStructsMap: Map<string, StateSnapshotStruct>;
+    //map [stateSnapshotHash] => [fork count, block height]
+    private hashToStateSnapshotKeyMap: Map<BytesLike, string>;
+    //could be replaced with a key, but we would need some computation
+    private latestOnChainStateSnapshot: {
+        stateSnapshot: StateSnapshotStruct;
+        timestamp: number;
+    };
+
+    private joinChannelBlockHashesMap: Map<string, BytesLike>;
+    //TODO: check if BytesLike is more efficient than string
+    // we can either store the hash or the key to the joinChannelBlockHashesMap data
+    private latestJoinChannelBlockHash: BytesLike;
+
+    //map [blockHash] => ExitChannelBlockStruct
+    private exitChannelBlockMap: Map<string, ExitChannelBlockStruct>;
+    //TODO: check if BytesLike is more efficient than string
+    // we can either store the hash or the key to the exitChannelBlockHashesMap data
+    private latestExitChannelBlockHash: BytesLike;
+
+    private totalDeposits: BalanceStruct;
+    private totalWithdrawals: BalanceStruct;
```

Comment:
Partition this further into smaller modules BlockStorage, SnapshotStorage etc... Then this class can be the StorageManager that exposes them and uses them internally - that way we can clearly see how you do storage for different components instead of this file having all the maps

Reply:
> Partition this further into smaller modules BlockStorage, SnapshotStorage etc... Then this class can be the StorageManager that exposes them and uses them internally - that way we can clearly see how you do storage for different components instead of this file having all the maps

yeah i planned to do a refactoring into smaller modules once i have the logic done

Reply:
Ah ok than I have to look at the logic now - wanted to look at it in isolation

Reply:
refactoring done